lib/ramble/ramble/language/application_language.py

```diff
@@ -165,20 +194,45 @@ def workload_variable(name, default, description, values=None, workload=None,
     """
 
     def _execute_workload_variable(app):
-        all_workloads = ramble.language.language_helpers.require_definition(workload,
-                                                                            workloads,
-                                                                            app.workloads,
-                                                                            'workload',
-                                                                            'workloads',
-                                                                            'workload_variable')
-
-        for wl_name in all_workloads:
-            app.workloads[wl_name].add_variable(
-                ramble.workload.WorkloadVariable(
-                    name, default=default, description=description,
-                    values=values, expandable=expandable
+        if (workload is not None or workloads is not None) and workload_group is None:
+            all_workloads =  \
+                ramble.language.language_helpers.require_definition(workload,
+                                                                    workloads,
+                                                                    app.workloads,
+                                                                    'workload',
+                                                                    'workloads',
+                                                                    'workload_variable')
+
+            for wl_name in all_workloads:
+                app.workloads[wl_name].add_variable(
+                    ramble.workload.WorkloadVariable(
+                        name, default=default, description=description,
+                        values=values, expandable=expandable
+                    )
                 )
-            )
+        elif workload is None and workloads is None and workload_group is not None:
+            # Find workload group, and iterate it
+            workload_group_list = app.workload_groups[workload_group]
+
+            # TODO: better place to init this?
+            if workload_group not in app.workload_group_vars:
+                app.workload_group_vars[workload_group] = []
+
+            # Track which vars we add to, to allow us to re-apply during inheritance
+
+            # TODO: dry this add with above?
+            workload_var = ramble.workload.WorkloadVariable(
```

Comment:
Maybe we should make a copy method (or clone?) on WorkloadVariable, so we can create one initially, and copy it into the different workloads.